### ISSUE TITLE
Staging Review finding: Keyboard focus lost between pages

### ISSUE THREAD

**Issue by shiragoodman** [department-of-veterans-affairs/va.gov-team#86829]
_Need help? Please review [how to read a Staging Review ticket](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/anatomy-of-a-staging-review-issue-ticket). Tag `@platform-governance-team-members` on Slack if you need further assistance._

## Product Information
**Team:** VYE Team
**Product:** Verify Your Enrollment
**Feature:** Verify Your Enrollment

## Findings details
**VA.gov Experience Standard - issue:** [User can't determine next steps in a flow.](https://design.va.gov/about/experience-standards/#user-can-determine-next-steps-in-a-flow)
**VA.gov Experience Standard - category:** [Findability](https://design.va.gov/about/experience-standards/#category-findability)
**Launch-blocking:** Yes
**Design System review:** No
**Collab Cycle Reviewer:** @briandeconinck (Accessibility)

### Description

There are a few places where keyboard focus is lost when transitioning through parts of the product.

Keyboard focus determines what the next tab stop is for keyboard users, and what will be the next content announced to screen reader users. On a "typical" web page, focus automatically moves to the top of the document when navigating from page to page. In a React app like this, more assertive focus management is required since the browser never reloads.

When focus is lost (meaning, focus isn't set to any element and the element that it was on disappears), this impacts:
- Keyboard users, who press [tab] and are taken to somewhere they don't expect.
- Screen reader users, who don't hear any confirmation that anything has happened after following the link.

Places where I'm losing focus:
- On [/mgib-enrollments/](https://staging.va.gov/education/verify-school-enrollment/mgib-enrollments/) when following the "Start enrollment verification" link to [/mgib-enrollments/verify-information/](https://staging.va.gov/education/verify-school-enrollment/mgib-enrollments/verify-information/)
- On [/mgib-enrollments/](https://staging.va.gov/education/verify-school-enrollment/mgib-enrollments/) when following the "Manage your Montgomery GI Bill benefits information" link to [/mgib-enrollments/benefits-profile/](https://staging.va.gov/education/verify-school-enrollment/mgib-enrollments/benefits-profile/)
- On [/mgib-enrollments/verify-information/](https://staging.va.gov/education/verify-school-enrollment/mgib-enrollments/verify-information/) when submitting an enrollment verification
- On [/mgib-enrollments/benefits-profile/](https://staging.va.gov/education/verify-school-enrollment/mgib-enrollments/benefits-profile/) when editing contact information
- On [/mgib-enrollments/benefits-profile/](https://staging.va.gov/education/verify-school-enrollment/mgib-enrollments/benefits-profile/) when adding/updating direct deposit information

<details><summary>Link, screenshot or steps to recreate</summary>

Example of focus being lost:

https://github.com/department-of-veterans-affairs/va.gov-team/assets/6844775/3aa91075-01b7-427d-a7f5-70a111bdba32

In the video, note that after following the link, screen reader users hear the loading indicator announced but never hear anything to indicate the new page has finished loading.

</details>

## Recommended action

On [/mgib-enrollments/](https://staging.va.gov/education/verify-school-enrollment/mgib-enrollments/) when following the "Start enrollment verification" link to [/mgib-enrollments/verify-information/](https://staging.va.gov/education/verify-school-enrollment/mgib-enrollments/verify-information/), focus should go to the H1 on the /verify-information/ page.

On [/mgib-enrollments/](https://staging.va.gov/education/verify-school-enrollment/mgib-enrollments/) when following the "Manage your Montgomery GI Bill benefits information" link to [/mgib-enrollments/benefits-profile/](https://staging.va.gov/education/verify-school-enrollment/mgib-enrollments/benefits-profile/), focus should go to the H1 on the /benefits-profile/ page.

On [/mgib-enrollments/verify-information/](https://staging.va.gov/education/verify-school-enrollment/mgib-enrollments/verify-information/) when submitting your enrollment verification, focus should go to the success alert (or `div` containing the alert) on the subsequent page. If there are any error states after submit, focus should likewise go to those alerts.

On [/mgib-enrollments/benefits-profile/](https://staging.va.gov/education/verify-school-enrollment/mgib-enrollments/benefits-profile/) when updating your contact information:
- When starting the edit workflow, focus should go to the "Change mailing address" H3 to indicate the start of the editing flow
- Successfully updating your contact information should send focus to the success alert (or a `div` containing the alert) so that users receive a confirmation that the action is complete
- Can't verify your address message should send focus to the warning alert (or a `div` containing the alert) so that users hear what the problem is
- Canceling editing should send focus back to the edit button

On [/mgib-enrollments/benefits-profile/](https://staging.va.gov/education/verify-school-enrollment/mgib-enrollments/benefits-profile/) when updating your direct deposit information:
- When starting the add/change flow, focus should go to the "Add new account" H3
- Successfully updating banking information should send focus to the success alert
- Any error states (I didn't encounter any in testing) should send focus to the warning or error alert
- Canceling editing should send focus back to the Add or change account button

## References

- **Accessibility Defect Severity:** [1: Critical. Must be fixed before launch.](https://depo-platform-documentation.scrollhelp.site/developer-docs/accessibility-defect-severity-rubric#Accessibilitydefectseverityrubric-a11y-defect-1)
- **WCAG Success Criteria:** N/A
- **Modality:** Screen reader
- **Design System Component:** N/A
- **Design System Pattern or Template:** N/A
- **Design System Foundation:** N/A
- **Content Style Guide:** N/A
- **Context:** N/A

---

### Next Steps for the VFS Team

- [ ] **Questions?** For the most timely response, comment on Slack in your team channel tagging `@platform-governance-team-members` with any questions or to get help validating the issue.
- [ ] **Ticket ownership:** If you believe that this issue is out of scope, not your team's responsibility, or a Design System issue, comment and tag `@platform-governance-team-members` on your team channel in Slack to provide an explanation and who you believe is responsible. The Governance team will follow up.
- [ ] **Close this ticket** when the issue has been resolved or validated by your Product Owner.

**Comment by wafimohamed** [comment]
fixed issue with losing focus when navigating to new page
https://github.com/department-of-veterans-affairs/vets-website/pull/30565

**Comment by jsimonVA** [comment]
Update has been made and is in staging.
